hangupsbot/plugins/cleverbot.py

```diff
@@ -251,66 +29,32 @@ def _handle_incoming_message(bot, event, command):
     percentage = bot.get_config_suboption(event.conv_id, 'cleverbot_percentage_replies')
 
     if randrange(0, 101, 1) < float(percentage):
-        text = yield from cleverbot_ask(event.conv_id, event.text)
-        if text:
-            yield from bot.coro_send_message(event.conv_id, text)
-
+        yield from chat(bot, event)
 
 def chat(bot, event, *args):
-    """chat to Cleverbot"""
+    """ Chat to Cleverbot. Usage: /bot chat hi cleverbot! """
+    if not cw:
+        api_key = bot.get_config_option("cleverbot_api_key")
+        if not api_key:
+            response = "API key not defined: config.cleverbot_api_key"
+            logger.error(response)
+            yield from bot.coro_send_message(event.conv_id, response)
+            return
+        else:
+            global cw
+            cw = CleverWrap(api_key)
     if args:
-        text = yield from cleverbot_ask(event.conv_id, ' '.join(args))
-        if not text:
-            text = _("<em>Cleverbot is silent</em>")
-
+        input_text = " ".join(args)
     else:
-        text = _("<em>you have to say something to Cleverbot</em>")
-
-    yield from bot.coro_send_message(event.conv_id, text)
+        input_text = event.text
+    response = cw.say(input_text)
```

Comment:
I don't think it'd be that memory intensive to hold multiple cw instances. Is this "feature" really necessary though? If you tell the bot something in one channel, shouldn't it "remember" it when you ask it in another? Maybe we should make this behaviour configurable...